dashboard/app/models/ability.rb

```diff
@@ -205,7 +205,7 @@ def initialize(user)
           !script.old_professional_learning_course?
         end
         can [:read, :find], :regional_partner_workshops
-        can [:new, :create, :read, :update], TEACHER_APPLICATION_CLASS, user_id: user.id
+        can [:new, :create, :show, :update], TEACHER_APPLICATION_CLASS, user_id: user.id
```

Comment:
This is a subtle one!

In addition to Rails 6.1 support, this version of CanCanCan also added support for single-table inheritance, meaning that permissions can now cascade to their descendants. Unfortunately, this revealed what I think is a bug in our implementation logic; our intent here is to grant teachers view permissions to the teacher application model. That model [is based on](https://github.com/code-dot-org/code-dot-org/blob/ca45703d9a0cc838d3b0bd30ce261e5036ebb137/dashboard/app/models/pd/application/teacher_application.rb#L39) our [`ApplicationBase` model](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/application/application_base.rb), the permissions of which [also control the route that lets you query all applications](https://github.com/code-dot-org/code-dot-org/blob/ca45703d9a0cc838d3b0bd30ce261e5036ebb137/dashboard/app/controllers/pd/application_dashboard_controller.rb#L3).

We don't actually want teachers to be able to access _that_ route, but because [`read` is an alias for `index` plus `show`](https://github.com/CanCanCommunity/cancancan/blob/develop/docs/define_check_abilities.md#can-actions), we are implicitly (and I think accidentally) relying on single-table inheritance not working here to avoid this permission cascading up to the application dashboard controller.

The simple fix is to be more targeted in the permissions we grant.